frontend/front/src/pages/Surveyor/houseProfile/HouseProfile.js

```diff
@@ -41,27 +50,50 @@ const HouseProfile = () => {
 
   const submitSurvey = useCallback(
     async (responses, surveyId) => {
-      return await addSurveyVisit({
-        responses,
-        homeId,
-        surveyId,
-        // TODO: probably remove this and handle on the back end
-        date: new Date().toISOString(),
+      const surveyAnswers = {};
+      Object.entries(responses).forEach(([key, value]) => {
```

Comment:
Since you have this in two places it might be good to factor it out into a shared util function